### ISSUE TITLE
Preserve Helm Parameters case

### ISSUE THREAD

**Issue by bryopsida** [eclipse/jkube#2356]
### Component

JKube Kit

### Is your enhancement related to a problem? Please describe

As a user I would like to be able to set/override default values of included helm dependencies as defaults in my generated  `Values.yaml`. This can be problematic when that sub chart uses camelCasing.  For example when running `k8sResource k8sHelm` with the gradle plugin, `postgres.someValue` ends up as `postgres.somevalue` in the generated `Values.yaml` file. This is especially problematic because helm best practices suggest camelCasing: https://helm.sh/docs/chart_best_practices/values/#naming-conventions which means most values have this issue.

### Describe the solution you'd like

I would like to see the original case provided in Helm.Parameters to be preserved in the generated `Values.yaml`. To not break anyone who might already have helm parameters with mixed case currently this would be an opt in toggle to preserve case.

### Describe alternatives you've considered

#2200 could solve part of the problem as well as long as the case is preserved from the provided `Values.yaml` file.

**Comment by rohanKanojia** [comment]
@bryopsida : I see you've got a PR related to helm open. Do you plan to work on this as well? Shall I assign it to you?

**Comment by bryopsida** [comment]
@rohanKanojia yes, I was planning on taking this on as well.

**Comment by manusa** [comment]
I can't really find a reason as to why we're transforming the parameter names to lower-case.

The only reason I can think of is to make them case-insensitive.

The change was added as part of #116 when the Helm logic was extracted from the Mojo to a separate module in JKube Kit.

Fabric8 Maven Plugin's original code doesn't include any case transformation either:

https://github.com/fabric8io/fabric8-maven-plugin/blob/89af88323fa685d8b298ba4b2cbc415b9b2911f0/plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/HelmMojo.java#L682-L684

In conclusion, I think it's safe to preserve the case-sensitivity and I don't really see the need for adding an additional flag. I'll refactor #2369 accordingly.
